packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -399,17 +399,41 @@ class WebpackCLI {
             },
         ];
 
+        const normalizeType = (type, simpleType) => {
+            if (type === 'string') {
+                return String;
+            } else if (type === 'number') {
+                return Number;
+            } else if (type === 'boolean' || type === 'reset') {
+                return Boolean;
+            } else if (type === 'RegExp') {
+                return RegExp;
+            } else if (type === 'enum') {
+                return normalizeType(simpleType);
+            }
```

Comment:
Tackling this `enum` type is tricky. Open to suggestions for improvement.

Reply:
We have more types https://github.com/webpack/webpack/blob/master/lib/cli.js#L42

Reply:
for enum I think you need to run `typeof` and collect all possible types